src/components/Loginform.jsx

```
@@ -0,0 +1,153 @@
+    const isLoading = useSelector((state) => state.login.isLoading);
+    const dispatch = useDispatch();
+    const [showPassword, setShowPassword] = useState(false);
+    const [recaptchaToken, setRecaptchaToken] = useState("");
+
+    const validateField = (field, value) => {
+        let err = "";
+        if (field === "email") {
+            if (!value) err = "Email is required";
+            else if (!/^\S+@\S+\.\S+$/.test(value)) err = "Email is invalid";
+        }
+        if (field === "password") {
+            if (!value) err = "Password is required";
+        }
+        return err;
+    };
+
+    const handleChange = (e) => {
+        const { name, value, type, checked } = e.target;
+        const fieldValue = type === "checkbox" ? checked : value;
+        dispatch(updateLoginForm(name, fieldValue));
+
+        const errorMsg = validateField(name, fieldValue);
+        dispatch(
+            setLoginErrors({
+                ...errors,
+                [name]: errorMsg || undefined,
+            })
+        );
+    };
+
+    const validateAll = () => {
+        let currErrors = {};
+        Object.keys(form).forEach((key) => {
+            const errorMsg = validateField(key, form[key]);
+            if (errorMsg) currErrors[key] = errorMsg;
+        });
+        dispatch(setLoginErrors(currErrors));
+        return Object.keys(currErrors).length === 0;
+    };
+
+    const handleRecaptchaChange = (token) => {
+        dispatch(updateLoginForm("recaptchaToken", token));
+    };
+
+    const handleSubmit = async (e) => {
+        e.preventDefault();
+        console.log(form);
+
+        if (!form.recaptchaToken) {
+            alert("Please complete the CAPTCHA.");
+            return;
+        }
+
+        if (validateAll()) {
+            dispatch(setLoginLoading(true));
+            try {
+                await onSubmit(form);
+                dispatch(resetLoginForm());
+            } catch (err) {
```

empty catch block - handle or log err
eg - `console.error("Error during form submission:", err);`